wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -1128,11 +1128,19 @@ private function make_sqlite_field_definition( $field ) {
 		 * This mode allows the use of `NULL` when NOT NULL is set on a column that falls back to DEFAULT.
 		 * SQLite does not support this behavior, so we need to add the `ON CONFLICT REPLACE` clause to the column definition.
 		 */
-		if (null !== $field->default && $field->not_null) {
+		if ($field->not_null) {
 			$definition .= ' ON CONFLICT REPLACE';
 		}
-		if ( null !== $field->default ) {
+		if (null === $field->default) {
+			$definition .= ' DEFAULT NULL';
+		} else if (false !== $field->default) {
 			$definition .= ' DEFAULT ' . $field->default;
+		} else if ($field->not_null) {
+			if ('text' === $field->sqlite_data_type) {
+				$definition .= ' DEFAULT \'\'';
+			} else {
+				$definition .= ' DEFAULT 0';
+			}
```

Comment:
@adamziel what do you think about this solution? 
Instead of rewriting update queries with COALESCE we could always set a default.

Reply:
This will allow both inserting NULL values and updating existing rows to NULL values. MySQL rejects inserting NULL values and only allows updating. I think it's fine for now, let's just create a new issue to track rejecting inserts in the future.

Reply:
https://github.com/WordPress/sqlite-database-integration/issues/97